modules/apps/portal-workflow/portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/util/WorkflowDefinitionContentUtil.java

```diff
@@ -50,6 +51,8 @@ public static String toJSON(String content) throws WorkflowException {
 			DocumentBuilder documentBuilder =
 				documentBuilderFactory.newDocumentBuilder();
 
+			content = StringUtil.replace(content, '&', "&amp;");
```

Comment:
Hey @p-albuquerque we have to iterate over the XML file,

you can use SAXReaderUtil to read the xml string and then iterate over the tree by using BFS for example.

````java
Document document = SAXReaderUtil.read(xml);

document.nodes();
````

According to https://www.w3.org/TR/xml/ in the section 2.4 Character Data and Markup 

> _the ampersand character (&) and the left angle bracket (<) must not appear in their literal form, except when used as markup delimiters, or within a [comment](https://www.w3.org/TR/xml/#dt-comment), a [processing instruction](https://www.w3.org/TR/xml/#dt-pi), or a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection)._

so you have to check other places different than those ones and escape as suggested by the document as well:

> _If they are needed elsewhere, they must be [escaped](https://www.w3.org/TR/xml/#dt-escape) using either [numeric character references](https://www.w3.org/TR/xml/#dt-charref) or the strings " &amp; " and " &lt; " respectively._

to do this you can perform a check like this:

````java
node instanceof CDATA
````

Thank you!